RangeRequest.bs

```diff
@@ -10,26 +10,218 @@ Editor: Chris Lilley, W3C, https://svgees.us/, w3cid 1438
+
+<h3 id="font-organization-compression">
+Compression</h2>
+
+A [=range-request optimized font=] should not use any kind of compression other than [[!RFC7932]] (commonly referred to as "Brotli") compression.
+
+If Brotli compression is used in a [=range-request optimized font=], it should use only one <a href="https://datatracker.ietf.org/doc/html/rfc7932#section-9.2">meta-block</a>.
+
+If Brotli compression is used in a [=range-request optimized font=], its one meta-block should have the <a href="https://datatracker.ietf.org/doc/html/rfc7932#section-9.2"><code>ISUNCOMPRESSED</code></a> bit set to 1.
+
+Issue: We can probably do way better than just disabling all compression.
+
+<h3 id="font-organization-table-ordering">
+Table Ordering</h2>
+
+No two tables in a [=range-request optimized font=] should share a tag name.
+
+A [=range-request optimized font=] should have only one of these tables:
+- The <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/cff"><code>CFF</code></a> table
+- The <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/cff2"><code>CFF2</code></a> table
+- The <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/glyf"><code>glyf</code></a> table
+
+The term <dfn>outline table</dfn> is used to describe the one table which is either a <code>CFF</code> table, <code>CFF2</code> table, or <code>glyf</code> table.
+
+The [=outline table=] data in a [=range-request optimized font=] should lie at the end of the file.
+
+If a <code>CFF</code> table exists, the <a href="https://wwwimages2.adobe.com/content/dam/acom/en/devnet/font/pdfs/5176.CFF.pdf">CharString</a> data should lie at the end of the <code>CFF</code> table.
+
+If a <code>CFF2</code> table exists, the <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/cff2#charStrings">CharString</a> data should lie at the end of the <code>CFF2</code> table.
+
+Issue: https://github.com/w3c/IFT/issues/28 What about font collections?
+
+<h3 id="font-organization-glyph-independence">
+Glyph Independence</h2>
+
+Note: The goal of this section is to make every glyph independent from each other.
+
+A [=range-request optimized font=] should not use <a href="https://developer.apple.com/fonts/TrueType-Reference-Manual/RM03/Chap3.html">TrueType instructions</a>.
+
+Note: Some rasterizer software does not even execute these instructions in the first place.
+
+A [=range-request optimized font=] should not use <a href="https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6glyf.html">Compound glyphs</a>.
+
+Note: Compound glyphs can be flattened by inlining their component glyphs to become additional contours.
+
```

Comment:
Fonts that were efficiently designed using composite glyphs _can be flattened_ but that can easily double the overall font file size. It seems counterproductive to require that we double the original font size only to make it eligible for a particular incremental font transfer method that, in the end, may not even offer a tangible benefit - a client may be better off downloading the whole original font from the get go.

Reply:
At minimum this should be tested out.  Did we study the expansion from flattening and other pre-processing steps?

Reply:
A while back, Monotype has made an effort to create a sizeable set of CJK fonts that would be suitable for OEM use (targeting the smallest storage size possible). They redesigned a [number of popular font families](https://catalog.monotype.com/font/oem-monotype/csong-tc/light-catt?keyword=CATT%20fonts) to employ as much of nested composite glyphs as possible, and released them under the CATT (Compact Asian TrueType) product name. IIRC, the size savings we've seen were quite good, e.g. an original flat design that needed ~9MB of storage would be reduced to ~4MB by utilizing composite glyph designs. I guess we can ask someone at Monotype to compare sizes of CATT fonts and their regular counterparts.

Reply:
> As for data, the PFE analysis used flattened fonts for simulating the range request method so for the fonts tested by the analysis we saw significant byte reductions for CJK despite the flattening operation.

I do remember WG discussions related specifically to CFF (comparing the sizes of subroutinized vs. de-subroutinized charstrings). The size variations we saw were about 10-15%,  and de-subroutinization with subsequent Brotli compression produced better results in terms of bytes transferred overall. We did use flattened TTF glyph outlines for evaluation but I don't remember if we actually compared flattened fonts to their composite counterparts.

Considering the fact that simple and composite TTF glyphs are easily distinguishable, font optimization process can rearrange their sequences by re-grouping simple glyphs and composite glyphs together, to make them continuous data blocks for range-request mechanism. I honestly don't see an immediate extra benefit of flattening composite CJK fonts because preprocessing can easily analyze composite glyphs, group all simple radicals into a continuous data block, and transfer them in a single range request. Composite glyphs would then refer to these simple glyphs, and considering that heavily composited CJK fonts reuse many of these simple radicals many times over - I don't think that flattening TTF outlines is truly necessary for range-request transfer.

Reply:
For continued discussion, see issue #58

Reply:
Because we have a dedicated issue for this, I'll resolve this for now, and add an issue in the PR with a link to the issue.